all-actors/src/main/java/org/sunbird/cert/actor/TemplateValidateActor.java

```diff
@@ -0,0 +1,106 @@
+        logger.info("onReceive method call start for operation " + operation);
+        if (JsonKey.VALIDATE_TEMPLATE.equalsIgnoreCase(operation)) {
+            validateTemplate(request);
+        }
+    }
+
+
+    private void validateTemplate(Request request) throws BaseException {
+        HTMLValidatorResponse validatorResponse;
+        String templateUrl = (String) request.getRequest().get(JsonKey.TEMPLATE_URL);
+        HTMLTemplateZip htmlTemplateZip = new HTMLTemplateZip(new LocalStore(certsConstant.getDOMAIN_URL()), templateUrl);
+
+        //download the file
+        try {
+            htmlTemplateZip.download();
+        } catch (StorageServiceException | IOException e) {
+            logger.info("exception while downloading " + e.getMessage());
+            throw new BaseException(IResponseMessage.INTERNAL_ERROR, e.getMessage(), ResponseCode.SERVER_ERROR.getCode());
+        }
+
+        //check if zip file downloaded or not ,if downloaded unzip
+        if (Boolean.TRUE.equals(htmlTemplateZip.isZipFileExists())) {
+            try {
+                htmlTemplateZip.unzip();
+                if (Boolean.TRUE.equals(htmlTemplateZip.isIndexHTMlFileExits())) {
+                    validatorResponse = validateHtml(htmlTemplateZip);
+                } else {
+                    throw new BaseException("INVALID_ZIP_FILE", MessageFormat.format(IResponseMessage.INVALID_ZIP_FILE, ":zip file format is invalid, unable to find file index.html"), ResponseCode.BAD_REQUEST.getCode());
+                }
+            } catch (IOException e) {
+                logger.info("exception while unzipping " + e.getMessage());
+                throw new BaseException("INVALID_ZIP_FILE", MessageFormat.format(IResponseMessage.INVALID_ZIP_FILE, ": exception while unzipping " + e.getMessage()), ResponseCode.BAD_REQUEST.getCode());
+            } finally {
+                htmlTemplateZip.cleanUp();
+            }
+        } else {
+            htmlTemplateZip.cleanUp();
+            throw new BaseException("INVALID_TEMPLATE_URL", MessageFormat.format(IResponseMessage.INVALID_TEMPLATE_URL, ": unable to download zip file , please provide valid url"), ResponseCode.BAD_REQUEST.getCode());
+        }
+
+        Response response = new Response();
+        response.getResult().put("response", validatorResponse);
+        sender().tell(response, getSelf());
+        logger.info("onReceive method call End");
+    }
+
+    private HTMLValidatorResponse validateHtml(HTMLTemplateZip htmlTemplateZip) {
+        HTMLValidatorResponse validatorResponse = new HTMLValidatorResponse();
+        String htmlContent;
+        try {
+            htmlContent = htmlTemplateZip.getTemplateContent();
+            HTMLTemplateValidator htmlTemplateValidator = new HTMLTemplateValidator(htmlContent);
+            Set<String> invalidVars = htmlTemplateValidator.validate();
+            if (invalidVars.isEmpty()) {
+                logger.info("template is valid");
```

Comment:
we must rely on the request id or so to identify which template is valid. Statements like this are pretty much less useful without contextual information. Nothing to do, but letting you recall your other efforts.